### PR TITLE
Warn user when saving contents of XQuery window with non-.xq* file extension

### LINKED ISSUE #126: Warn user when saving contents of XQuery window with non-.xq* file extension
In [a stackoverflow thread](http://stackoverflow.com/questions/38359614/improve-performance-of-query-with-range-indexes-in-exist-db/38361339), the user who posted the question and I determined that the reason his indexes weren't being applied was that he had saved the collection.xconf file from an XQuery window (started via "New XQuery") in eXide. As a result, the file's mimetype was not XML, eXide didn't offer to save the index in `/db/system`, and he was extremely confused. 

A warning should appear when saving the contents of an XQuery window with a non-`.xq*` filename, prompting the user to consider copying and pasting the file into a new window using the "New" button instead of the "New XQuery" window and selecting the appropriate file type.

### PR DESCRIPTION
**Pull request by marmoure**
Fixes https://github.com/eXist-db/eXide/issues/126.

warns user when trying to save contents of XQuery window with non-.xq* file extension and prevents it
suggests to use a new file if content is non-xquery

image

This open source contribution to the [eXide](https://github.com/eXist-db/eXide) project was commissioned by the Office of the Historian, U.S. Department of State, https://history.state.gov.

### CODE REVIEW

File: src/eXide.js [hunk 1]
```diff
@@ -398,7 +398,7 @@ eXide.app = (function(util) {
     					"Save": function() {
                             // force saving XQuery files
                             if(editor.getActiveDocument().isXQuery() && !/([^\s\\])*\.xq.?/.test(dbBrowser.getSelection().name)) {
-                                util.Dialog.warning("Failed to Save Document", "Your current file is an XQuery, but you are trying to save it with a non XQuery File Extension  (e.g. not .xq). If you intended this to be an XML file, or an other file type, then please copy and paste the content of your file into a New file that you create using the “New” button.");
+                                util.Dialog.warning("Failed to Save Document", "Your current file is an XQuery file, but you are trying to save it with a non-XQuery file extension (.xq, .xqm etc).  If you intended this to be another file type such as XML, copy and past the content into a New file created using the \"New\" button.");
```
Comment (joewiz) [inline src/eXide.js]: Looking good! A typo crept in here. "Past" should be corrected to "paste." Same in the other copy of the message.

**Review by joewiz — APPROVED** [review]
@marmoure This works perfectly for me in both contexts: save and save as. Thank you!

**Review by dizzzz — COMMENTED** [review]
lgtm

**Review by adamretter — APPROVED** [review]
LGTM

**Pull request merged**